receiver/k8sclusterreceiver/watcher.go

```diff
@@ -130,6 +149,28 @@ func (rw *resourceWatcher) onUpdate(oldObj, newObj interface{}) {
 	rw.syncMetadataUpdate(oldMetadata, newMetadata)
 }
 
+// waitForInitialInformerSync waits for initial cache sync, after the informers
+// are started up. This is achieved by resourceWatcher.informersHaveSynced. This
+// method is intended to be used by the ResourceEventHandler callbacks to ensure
+// that data sync does not start until the cache is synced as the events could be
+// received by the handler even before the informer caches being synced.
+func (rw *resourceWatcher) waitForInitialInformerSync() {
+	// Return immediately if already synced.
+	if rw.informersHaveSynced.Load() {
+		return
+	}
+
+	waitUntil := time.Now().Add(rw.initialCacheSyncTimeout)
+	for !rw.informersHaveSynced.Load() {
+		if time.Now().After(waitUntil) {
+			rw.logger.Warn("Initial sync of informers cache not yet completed. Try increasing 'initial_cache_sync_timeout'.")
```

Comment:
Why do we need this timeout? What if we just keep waiting for the cache as long as needed? We will not start updating dimensions until that, right? Isn't it desired behavior? Why would we need to send "corrupted" updates if we wasn't able to wait for the cache?

Reply:
I was thinking about this as well since we always want to make sure the data is accurate before starting to sync. I've removed this timeout.